paimon-core/src/main/java/org/apache/paimon/catalog/CachingCatalog.java

```diff
@@ -266,6 +266,14 @@ public List<PartitionEntry> listPartitions(Identifier identifier)
         return result;
     }
 
+    public void refreshPartitions(Identifier identifier) throws TableNotExistException {
+        if (partitionCache != null) {
+            partitionCache.invalidate(identifier);
```

Comment:
Why we need to invalidate it first?

Reply:
Just copied from sr, didn't think through. I'll delete it.